gnat2goto/driver/tree_walk.adb

```diff
@@ -3531,36 +3539,26 @@ package body Tree_Walk is
    end Do_Op_Minus;
 
    -------------------------
-   --      Do_Op_Or       --
-   -------------------------
-
-   function Do_Op_Or (N : Node_Id) return Irep is
-      LHS_Bool_Value : constant Irep := Do_Expression (Left_Opnd (N));
-      RHS_Bool_Value : constant Irep := Do_Expression (Right_Opnd (N));
-      Cast_LHS_To_Integer : constant Irep :=
-        Make_Op_Typecast (Op0 => LHS_Bool_Value,
-                          Source_Location => Sloc (N),
-                          I_Type => Make_Signedbv_Type (Make_Nil_Type, 32));
-      Cast_RHS_To_Integer : constant Irep :=
-        Make_Op_Typecast (Op0 => RHS_Bool_Value,
-                          Source_Location => Sloc (N),
-                          I_Type => Make_Signedbv_Type (Make_Nil_Type, 32));
-      R : constant Irep := Make_Op_Bitor (Lhs => Cast_LHS_To_Integer,
-                                          Rhs => Cast_RHS_To_Integer,
-                                          Source_Location => Sloc (N),
-                                          I_Type =>
-                                            Get_Type (Cast_LHS_To_Integer));
-   begin
-      return Make_Op_Typecast (Op0 => R,
-                               Source_Location => Sloc (N),
-                               I_Type => Make_Bool_Type);
-   end Do_Op_Or;
-
-   -------------------------
-   --      Do_Op_And      --
+   --      Do_Bit_Op      --
    -------------------------
 
-   function Do_Op_And (N : Node_Id) return Irep is
+   --  We identified that the constructor for operator `or` and operator
+   --  `and` where pretty much the same, with the only difference being
+   --  the constructor being called. So to avoid needless duplication,
+   --  we simplified it to a single function that does the same thing,
+   --  and just calls the appropriate constructor via a function pointer.
+   --
+   --  This produces the following code in pseudocode
+   --    A or B  (assume A => True and B => True)
```

Comment:
I feel this is a bit confusing - this looks like it'd be the pseudocode, but it's actually still part of the 'preamble' so to speak. How about 

> let A and B be `True` in the following example (pseudocode)